### ISSUE TITLE
There is no support for SSH options from ssh_config

### ISSUE THREAD

**Issue by click0** [jimsalterjrs/sanoid#518]
```
root@master:~ # uname -a
FreeBSD master.domain.com 12.1-STABLE FreeBSD 12.1-STABLE #0 r358094M: Wed Feb 26 21:34:58 EET 2020     root@core.domain.com:/usr/obj/usr/src/amd64.amd64/sys/core.domain.com.3  amd64
```
```
root@master:~/.ssh# cat /root/.ssh/ssh_config
host 192.168.66.104
    user                zfs-replication
    IdentityFile        ~/.ssh/id_ecdsa-zfs-replication
```

I am trying to start synchronization:
```
syncoid --debug --compress zstd-fast --source-bwlimit 28m --no-clone-rollback zroot/test 192.168.66.104:zroot/test
```
And I get the output:

```
DEBUG: SSHCMD: ssh
DEBUG: checking availability of zstd on source...
DEBUG: checking availability of zstd on target...
DEBUG: checking availability of zstd on local machine...
DEBUG: checking availability of mbuffer on source...
DEBUG: checking availability of mbuffer on target...
DEBUG: checking availability of pv on local machine...
DEBUG: checking availability of zfs resume feature on source...
DEBUG: checking availability of zfs resume feature on target...
WARN: ZFS resume feature not available on target machine - sync will continue without resume support.
DEBUG: syncing source zroot/test to target 192.168.66.104:zroot/test.
DEBUG: getting current value of syncoid:sync on zroot/test...
  zfs get -H syncoid:sync 'zroot/test'
DEBUG: checking to see if 192.168.66.104:zroot/test on  is already in zfs receive using  ps -Ao args= ...
DEBUG: checking to see if target filesystem exists using "  zfs get -H name '192.168.66.104:zroot/test' 2>&1 |"...
DEBUG: getting list of snapshots on zroot/test using   zfs get -Hpd 1 -t snapshot guid,creation 'zroot/test' |...
DEBUG: creating sync snapshot using "  zfs snapshot 'zroot/test'@syncoid_master.domain.com_2020-03-04:04:01:14
"...
DEBUG: target 192.168.66.104:zroot/test does not exist.  Finding oldest available snapshot on source zroot/test ...
DEBUG: getting estimated transfer size from source  using "  zfs send  -nP 'zroot/test@syncoid_master.domain.com_2020-03-04:03:45:50' 2>&1 |"...
DEBUG: sendsize = 12912
INFO: Sending oldest full snapshot zroot/test@syncoid_master.domain.com_2020-03-04:03:45:50 (~ 12 KB) to new target filesystem:
DEBUG:  zfs send  'zroot/test'@'syncoid_master.domain.com_2020-03-04:03:45:50' | mbuffer -R 28m -q -s 128k -m 16M 2>/dev/null | pv -s 12912 |  zfs receive   -F '192.168.66.104:zroot/test'
DEBUG: checking to see if 192.168.66.104:zroot/test on  is already in zfs receive using  ps -Ao args= ...
cannot open '192.168.66.104:zroot/test': dataset does not exist
cannot receive new filesystem stream: dataset does not exist
53.5KiB 0:00:00 [6.01MiB/s] [=================================================================================] 424%
CRITICAL ERROR:  zfs send  'zroot/test'@'syncoid_master.domain.com_2020-03-04:03:45:50' | mbuffer -R 28m -q -s 128k -m 16M 2>/dev/null | pv -s 12912 |  zfs receive   -F '192.168.66.104:zroot/test' failed: 256 at /usr/local/bin/syncoid line 464.
```

**Comment by phreaker0** [comment]
There is already an PR which will also fix this: #374

**Comment by phreaker0** [comment]
#374 was merged.
